opslevel/resource_opslevel_check_tool_usage.go

```diff
@@ -43,13 +44,14 @@ type CheckToolUsageResourceModel struct {
 	Notes       types.String `tfsdk:"notes"`
 	Owner       types.String `tfsdk:"owner"`
 
-	ToolCategory         types.String    `tfsdk:"tool_category"`
-	ToolNamePredicate    *PredicateModel `tfsdk:"tool_name_predicate"`
-	ToolUrlPredicate     *PredicateModel `tfsdk:"tool_url_predicate"`
-	EnvironmentPredicate *PredicateModel `tfsdk:"environment_predicate"`
+	ToolCategory         types.String `tfsdk:"tool_category"`
+	ToolNamePredicate    types.Object `tfsdk:"tool_name_predicate"`
+	ToolUrlPredicate     types.Object `tfsdk:"tool_url_predicate"`
+	EnvironmentPredicate types.Object `tfsdk:"environment_predicate"`
 }
 
 func NewCheckToolUsageResourceModel(ctx context.Context, check opslevel.Check, planModel CheckToolUsageResourceModel) CheckToolUsageResourceModel {
+	var diags diag.Diagnostics
```

Comment:
I think we are appending to this variable but it is never actually used for anything?